Update Core Workflow (#31)
ombulabs/claude-code_rails-upgrade-skill: rails-upgrade/SKILL.md

Author: Why Common Request Patterns was trimmed?

Pattern 1/2/3 restated the Core Workflow step-by-step per entry point (full upgrade, multi-hop, analysis-only).
Same drift problem as `High-Level Workflow`: every step change had to be made in four places.

Trimmed each pattern to one short paragraph that names which steps apply for that entry point (1–9 for full upgrade, 1+2+6 for analysis-only, etc.).
The actual step content stays in Core Workflow. Patterns now answer "where do I enter?" not "what are the steps?"

- See `workflows/test-suite-verification-workflow.md`.

### Step 2: Verify Latest Patch Version
- Before hopping minor/major, verify the app is on the latest patch of its current Rails series.
- Read `Gemfile.lock` for the exact current Rails version.
- Compare against the latest patch for that series:
  - **EOL series (≤ 7.1):** static table in `references/multi-hop-strategy.md`.
  - **Active series (≥ 7.2):** query RubyGems at runtime (commands in the same reference).
- If not on latest patch: update the Gemfile, run `bundle update rails`, run tests, deploy the patch, then proceed.
- **Why:** patch releases contain security fixes, bug fixes, and additional deprecation warnings that make the next hop safer.

### Step 3: Resolve Deprecation Warnings
- **DELEGATE** to `workflows/deprecation-resolution-workflow.md` (content lands in a follow-up PR).
- Methodology: regex search, `next_rails` deprecation tooling, `synvert-ruby`, prefer backward-compatible fixes over `NextRails.next?` conditionals when possible.
- Goal: the deprecation list from Step 1 is empty (or down to known, justified exceptions) before introducing the next Rails version.

> ### Step 4: Review Ruby Compatibility

Author: Current bullets (minimum-Ruby check) more in upcoming PRs

- Collect findings with file:line references. See `workflows/direct-detection-workflow.md`.
- Generate the Comprehensive Upgrade Report and `app:update` Preview from actual findings (templates in `templates/`).

### Step 7: Fix Broken Build
Fix-order discipline (content lands in a follow-up PR):
- **Errors before failures** — errors can mask failures.
- **Unit → integration → system** — a broken unit suite produces noise everywhere else.
- **RSpec:** Models → Services → Mailers → Helpers → Controllers → Integration → System.
- **Minitest:** Models → Mailers → Helpers → Controllers → Integration → System.

Use `NextRails.next?` for code that must work on both versions (DELEGATE to dual-boot for patterns). Do not fix deprecations printed by the next version here — those belong in the next upgrade's Step 3.

### Step 8: Smoke Test
Broader than `rails runner`. On `Gemfile.next` (checklist expanded in a follow-up PR):
- Open a Rails console.
- Start the Rails server.
- Start background workers (sidekiq, sucker_punch, delayed_job) — check `Procfile` for what's expected.
- Run `assets:precompile` in production mode.

> ### Step 9: Remove Dual-Boot

Author: Current bullets (minimum to do with Dual Boot) more in upcoming PRs

---

## Core Workflow

Follows the FastRuby.io methodology: current-version deprecations are the primary signal for what breaks next version, fix the build in dependency order, and treat `load_defaults` alignment as optional post-upgrade work.

### Step 1: Run Test Suite
- Run the existing test suite on the current Rails. All tests must pass before proceeding.
- Ensure deprecation warnings are NOT silenced in the test environment before running. Reconfigure to `:stderr` or `:log` if needed so Step 3 has the full list.
- Run `bundle exec rspec` or `bundle exec rails test`.
- If tests fail, stop and help fix them first.

> - Record test count, coverage, and the deprecation-warning list as baseline.

Reviewer: @JuanVqz I think this sounds like too much. I'd focus on just the deprecation warnings. 

```suggestion
- Record the list of deprecation warnings as a baseline.
```

Author: Done!

> - See `workflows/test-suite-verification-workflow.md`.

Reviewer: @JuanVqz This doesn't exist: 

> `workflows/test-suite-verification-workflow.md`

Did you commit all your changes?

Author: Fixed.

### Step 2: Verify Latest Patch Version
- Before hopping minor/major, verify the app is on the latest patch of its current Rails series.
- Read `Gemfile.lock` for the exact current Rails version.
- Compare against the latest patch for that series:
  - **EOL series (≤ 7.1):** static table in `references/multi-hop-strategy.md`.
  - **Active series (≥ 7.2):** query RubyGems at runtime (commands in the same reference).
- If not on latest patch: update the Gemfile, run `bundle update rails`, run tests, deploy the patch, then proceed.
- **Why:** patch releases contain security fixes, bug fixes, and additional deprecation warnings that make the next hop safer.

### Step 3: Resolve Deprecation Warnings

> - **DELEGATE** to `workflows/deprecation-resolution-workflow.md`.

Reviewer: This file doesn't exist: `workflows/deprecation-resolution-workflow.md`

Author: Fixed

---

> ## High-Level Workflow

Author: Why High-Level Workflow was removed?

This section duplicated Core Workflow with more verbose code-block formatting — same step numbers, same actions, just restated. It accounted for ~105 lines.
Keeping both means every future step change has to be made in two places, and they'll drift.

Core Workflow is the single source of truth now; detail goes into `workflows/<name>-workflow.md` per step, loaded on demand via progressive disclosure.

Net: 547 → 371 lines in SKILL.md, back under the 500-line limit.
